runtime/src/lib.rs

```diff
@@ -130,6 +135,65 @@ construct_runtime!(
     }
 );
 
+#[cfg(feature = "with-staking")]
+construct_runtime!(
+    pub enum Runtime where
+        Block = Block,
+        NodeBlock = nodle_chain_primitives::Block,
+        UncheckedExtrinsic = UncheckedExtrinsic
+    {
+        // System
+        System: frame_system::{Module, Call, Storage, Config, Event<T>},
+        Timestamp: pallet_timestamp::{Module, Call, Storage, Inherent},
+        Indices: pallet_indices::{Module, Call, Storage, Config<T>, Event<T>},
+        Balances: pallet_balances::{Module, Call, Storage, Config<T>, Event<T>},
+        TransactionPayment: pallet_transaction_payment::{Module, Storage},
+        RandomnessCollectiveFlip: pallet_randomness_collective_flip::{Module, Call, Storage},
+
+        // Consensus
+        Babe: pallet_babe::{Module, Call, Storage, Config, Inherent, ValidateUnsigned},
+        Grandpa: pallet_grandpa::{Module, Call, Storage, Config, Event, ValidateUnsigned},
+        Authorship: pallet_authorship::{Module, Call, Storage, Inherent},
+        ImOnline: pallet_im_online::{Module, Call, Storage, Event<T>, ValidateUnsigned, Config<T>},
+        Offences: pallet_offences::{Module, Call, Storage, Event},
+        Staking: pallet_curveless_staking::{Module, Call, Config<T>, Storage, Event<T>, ValidateUnsigned},
```

Comment:
Shall we handle this like Acala does to avoid including the pallet if this isn't necessary? https://github.com/AcalaNetwork/Acala/blob/c34cce131e0f9fbd69f2304365a0124a785ca01d/runtime/mandala/src/lib.rs#L1778 (they do it to differentiate between compiling as a parachain or as a standalone chain)

Reply:
Yes resolved the messy part :+1: